### PR TITLE
Expansion of viewport tooltips

### PR DESCRIPTION
**Pull request by RoqueDeicide**
## Motivation / Problem

More information could be displayed with viewport tooltips.

## Description

All tooltip-related settings had been moved to the new "Tooltips" section under "Interface".

Town viewport tooltips can now:

1. Be set to always be on or off.
2. Display population counts, if gui.population_in_label setting is on.
3. Display local company rating, if there is one.

Industry viewport tooltips can now:

1. Be configured to always be off entirely, each section of the tooltip can also be set to off.
2. Display a comma-separated list of required cargoes for the industry.
3. Display a list of stockpiled cargoes, or simply add them to the above-mentioned list of required cargoes.
4. Display cargo suffixes supplied by callback 37.

Added:

1. Viewport tooltips for depots, showing the number of vehicles inside and, optionally, which ones are stopped, waiting or are consists without a leading engine.
2. Viewport tooltips for stations, showing the name of the station (similarly to town viewport tooltips, these can be configured to always be off, be on only if station names are hidden, or always be on), and/or a list of cargoes that had ever been delivered to the station by any means other then final delivery, with counts as well as their ratings.

All defaults had been set to be in line with previous implementation spirit:

1. Town viewport tooltips only show, when town names are hidden, same goes for station names.
2. Industry viewport tooltips will show the name and a list of produced cargo.
3. Depot viewport tooltips will show the number of vehicles inside.
4. Station viewport tooltips will show the cargo list.

## Checklist for review

Some things are not automated, and forgotten often. This list is a reminder for the reviewers.
* The bug fix is important enough to be backported? (label: 'backport requested')
* This PR touches english.txt or translations? Check the [guidelines](https://github.com/OpenTTD/OpenTTD/blob/master/docs/eints.md)
* This PR affects the save game format? (label 'savegame upgrade')
* This PR affects the GS/AI API? (label 'needs review: Script API')
    * ai_changelog.hpp, gs_changelog.hpp need updating.
    * The compatibility wrappers (compat_*.nut) need updating.
* This PR affects the NewGRF API? (label 'needs review: NewGRF')
    * newgrf_debug_data.h may need updating.
    * [PR must be added to API tracker](https://wiki.openttd.org/en/Development/NewGRF/Specification%20Status)

### CODE REVIEW

**Comment by JGRennison** [comment]
Thanks for this. This looks interesting. I will take a closer look at it later in the week but I've got some initial comments for now.

File: src/lang/extra/english.txt [hunk 1]
```diff
@@ -1406,11 +1400,30 @@ STR_FINANCES_REPAY_TOOLTIP_EXTRA                                :{BLACK}{STRING}
...
+
+STR_INDUSTRY_VIEW_NAME_TOOLTIP                                  :{STRING}
+STR_INDUSTRY_VIEW_REQUIRED_TOOLTIP_FIRST                        :{STRING}{RAW_STRING}
+STR_INDUSTRY_VIEW_REQUIRED_TOOLTIP_NEXT                         :, {STRING}{RAW_STRING}
+STR_INDUSTRY_VIEW_STOCKPILED_TOOLTIP                            :{CARGO_LONG} waiting{RAW_STRING}
+STR_INDUSTRY_VIEW_TRANSPORTED_TOOLTIP_EXTENSION                 :{CARGO_LONG}{RAW_STRING} ({COMMA}%)
+STR_INDUSTRY_VIEW_INFO_TOOLTIP                                  :{BLACK}{RAW_STRING}
+
+STR_DEPOT_VIEW_COUNT_TOOLTIP                                    :{COMMA} indside
+STR_DEPOT_VIEW_TOTAL_TOOLTIP                                    :Of {COMMA} inside:
```
Comment (JGRennison) [inline src/lang/extra/english.txt:1418]: These are misspelled. Also would it be worth expanding these a bit ({COMMA} vehicles inside)?

File: src/settings_gui.cpp [hunk 2]
```diff
@@ -1967,14 +1967,29 @@ static SettingsContainer &GetSettingsTree()
...
+			{
+				tooltips->Add(new SettingEntry("gui.hover_delay_ms"));
+				tooltips->Add(new ConditionallyHiddenSettingEntry("gui.instant_tile_tooltip", []() -> bool { return _settings_client.gui.hover_delay_ms != 0; }));
+				tooltips->Add(new SettingEntry("gui.town_name_tooltip_mode"));
+				tooltips->Add(new SettingEntry("gui.industry_tooltip_show"));
+				tooltips->Add(new ConditionallyHiddenSettingEntry("gui.industry_tooltip_show_name", []() -> bool { return !_settings_client.gui.industry_tooltip_show; }));
+				tooltips->Add(new ConditionallyHiddenSettingEntry("gui.industry_tooltip_show_required", []() -> bool { return !_settings_client.gui.industry_tooltip_show; }));
+				tooltips->Add(new ConditionallyHiddenSettingEntry("gui.industry_tooltip_show_stockpiled", []() -> bool { return !_settings_client.gui.industry_tooltip_show; }));
+				tooltips->Add(new ConditionallyHiddenSettingEntry("gui.industry_tooltip_show_stockpiled_as_required", []() -> bool { return !_settings_client.gui.industry_tooltip_show || _settings_client.gui.industry_tooltip_show_stockpiled || !_settings_client.gui.industry_tooltip_show_required; }));
```
Comment (JGRennison) [inline src/settings_gui.cpp:1985]: Is this right, disabling when industry_tooltip_show_stockpiled is true doesn't seem like the thing to do?

File: src/station_gui.cpp [hunk 3]
```diff
@@ -3096,3 +3096,51 @@ bool ShouldShowBaseStationViewportLabel(const BaseStation *bst)
...
+char *StationGetSpecialStringExternal(char *buff, int x, const char *last);
+
+bool GetStationViewportTooltipString(const TileIndex tile, char *buffer_position, const char *buffer_tail)
+{
+	const StationID station_id = GetStationIndex(tile);
+	const Station *station = Station::Get(station_id);
+
+	bool next = false;
+
+	if (_settings_client.gui.station_viewport_tooltip_name == STNM_ALWAYS_ON ||
```
Comment (JGRennison) [inline src/station_gui.cpp:3115]: When mixing && and || it's a good idea to use parentheses. Currently it is ambiguous what you mean with these.

File: src/settings_gui.cpp (hunk 2 above)
Comment (RoqueDeicide) [reply src/settings_gui.cpp:1985]: This is by design, stockpiled cargoes can only be shown 1 of 2 ways: either as required (default behavior) or as stockpiled. Showing them both ways simultaneously, I believe, would be redundant.

**Comment by JGRennison** [comment]
Thanks, I've merged this with some adjustments

**Pull request merged**